app/controllers/shared/api.php

```diff
@@ -49,44 +50,127 @@
+                break;
+            case $document->getCollection() === 'functions':
+                $queueForUsage
+                    ->addMetric(METRIC_FUNCTIONS, $value); // per project
+
+                if ($event === Database::EVENT_DOCUMENT_DELETE) {
+                    $queueForUsage
+                        ->addReduce($document);
+                }
+                break;
+            case $document->getCollection() === 'deployments':
+                $queueForUsage
+                    ->addMetric(METRIC_DEPLOYMENTS, $value) // per project
+                    ->addMetric(METRIC_DEPLOYMENTS_STORAGE, $document->getAttribute('size') * $value) // per project
+                    ->addMetric(str_replace(['{resourceType}', '{resourceInternalId}'], [$document->getAttribute('resourceType'), $document->getAttribute('resourceInternalId')], METRIC_FUNCTION_ID_DEPLOYMENTS), $value)// per function
+                    ->addMetric(str_replace(['{resourceType}', '{resourceInternalId}'], [$document->getAttribute('resourceType'), $document->getAttribute('resourceInternalId')], METRIC_FUNCTION_ID_DEPLOYMENTS_STORAGE), $document->getAttribute('size') * $value);// per function
+
+                break;
+            case $document->getCollection() === 'executions':
+                $queueForUsage
+                    ->addMetric(METRIC_EXECUTIONS, $value) // per project
+                    ->addMetric(str_replace('{functionInternalId}', $document->getAttribute('functionInternalId'), METRIC_FUNCTION_ID_EXECUTIONS), $value);// per function
+                break;
+            default:
+                break;
+        }
+    } catch (Throwable $error) {
+        if (!empty($logger)) {
+            $log = new Log();
+            $isProduction = App::getEnv('_APP_ENV', 'development') === 'production';
+            $log
+                ->setNamespace("appwrite-stats-api")
+                ->setServer(\gethostname())
+                ->setVersion(App::getEnv('_APP_VERSION', 'UNKNOWN'))
+                ->setType(Log::TYPE_ERROR)
+                ->setMessage($error->getMessage())
+                ->setAction('appwrite-queue-usage')
+                ->addTag('verboseType', get_class($error))
+                ->addTag('code', $error->getCode())
+                ->addExtra('event', $event)
+                ->addExtra('collection', $document->getCollection())
+                ->addExtra('file', $error->getFile())
+                ->addExtra('line', $error->getLine())
+                ->addExtra('trace', $error->getTraceAsString())
+                ->addExtra('detailedTrace', $error->getTrace())
+                ->addExtra('roles', \Utopia\Database\Validator\Authorization::$roles)
+                ->setEnvironment($isProduction ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+            $logger->addLog($log);
```

Comment:
We already have a logger method in this file with all the default params set. I don't think we need to reinitialise it